src/FillArrays.jl

```diff
@@ -50,6 +53,7 @@ rank(F::AbstractFill) = iszero(getindex_value(F)) ? 0 : 1
 IndexStyle(::Type{<:AbstractFill{<:Any,N,<:NTuple{N,Base.OneTo{Int}}}}) where N = IndexLinear()
 
 
+""" `Fill{T, N, Axes} <: AbstractFill{T, N, Axes}` (lazy `fill` with axes)"""
```

Comment:
This is simealtanteously too terse and too verbose: the `<: AbstractFill{T, N, Axes}` is unneeded because a user can call `supertype` if they want that information. Try to make it look like the docs for lazy types in base, e.g., 
```julia
help?> UnitRange
search: UnitRange AbstractUnitRange UnitUpperTriangular UnitLowerTriangular

  UnitRange{T<:Real}

  A range parameterized by a start and stop of type T, filled with elements
  spaced by 1 from start until stop is exceeded. The syntax a:b with a and b
  both Integers creates a UnitRange.

  Examples
  ≡≡≡≡≡≡≡≡≡≡

  julia> collect(UnitRange(2.3, 5.2))
  3-element Array{Float64,1}:
   2.3
   3.3
   4.3
  
  julia> typeof(1:10)
  UnitRange{Int64}
```

Reply:
Done.